vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/model/implementation/Serializer.java

```diff
@@ -0,0 +1,116 @@
+package org.semanticweb.vlog4j.core.model.implementation;
+
+import java.util.List;
+
+/*-
+ * #%L
+ * VLog4j Core Components
+ * %%
+ * Copyright (C) 2018 - 2019 VLog4j Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.semanticweb.vlog4j.core.model.api.Conjunction;
+import org.semanticweb.vlog4j.core.model.api.Constant;
+import org.semanticweb.vlog4j.core.model.api.DataSourceDeclaration;
+import org.semanticweb.vlog4j.core.model.api.DatatypeConstant;
+import org.semanticweb.vlog4j.core.model.api.ExistentialVariable;
+import org.semanticweb.vlog4j.core.model.api.Fact;
+import org.semanticweb.vlog4j.core.model.api.LanguageStringConstant;
+import org.semanticweb.vlog4j.core.model.api.Literal;
+import org.semanticweb.vlog4j.core.model.api.NamedNull;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Term;
+import org.semanticweb.vlog4j.core.model.api.UniversalVariable;
+import org.semanticweb.vlog4j.core.model.api.Variable;
+
+/**
+ * A utility class with static methods to obtain the correct parsable string
+ * representation of the different data models.
+ * 
+ * @author Ali Elhalawati
+ *
+ */
+public final class Serializer {
+
+	private Serializer() {
+
+	}
+
+	public static String getRuleString(Rule rule) {
+		return rule.getHead() + " :- " + rule.getBody() + ".";
```

Comment:
this calls `toString() `on the head and body `Conjunctions`, which in turn calls `getConjunctionString()`. We should not depend to the toString method in `Conjunction`, but only on the `Serializer` methods. That is why this method should return
`getConjunctionString(rule.getHead())+" :- " + getConjunctionString(rule.getBody())+"."`